### PR TITLE
storage: fix io error may let the chunk map bit always pending

### PR DESCRIPTION
**Pull request by kevinXYin**
For now , when we get an io error in do_fetch_chunks, we just return
without clearing the chunk map pending bit. This may cause the following
io always wait for the pending bits.

Signed-off-by: Xin Yin <yinxin.x@bytedance.com>

### CODE REVIEW

**Comment by yqleng1987** [comment]
@kevinXYin , a new test job has been submitted. Please wait in patience. The test job url: [https://tone.openanolis.cn/ws/nrh4nnio/test_result/13334](https://tone.openanolis.cn/ws/nrh4nnio/test_result/13334)

**Comment by yqleng1987** [comment]
@kevinXYin , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/13334>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/13334>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/13334>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/13334>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/13334>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/13334>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/13334>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

**Review by hsiangkao — APPROVED** [review]
LGTM, it'd better for others to take another look.

File: storage/src/cache/cachedfile.rs [hunk 1]
```diff
@@ -379,15 +379,20 @@ impl FileCacheEntry {
...
-                    eio!(format!(
-                        "do_fetch_chunk failed to persist stargz chunk, {:?}",
-                        e
-                    ))
-                })?;
-                self.chunk_map
-                    .set_ready_and_clear_pending(chunk.as_base())
-                    .unwrap_or_else(|e| error!("set stargz chunk ready failed, {}", e));
+                match Self::persist_chunk(&self.file, chunk.uncompress_offset(), &buf) {
+                    Ok(_) => {
```
Comment (jiangliu) [inline storage/src/cache/cachedfile.rs:383]: Seems there's more fundamental issue here:
the chunkmap bits have not been marked as pending for stargz case:(

Comment (kevinXYin) [reply storage/src/cache/cachedfile.rs:383]: emm,  yeah , I think we need further check and test for the stargz case. But I do not have environment to test it. How about reverting the change for the stargz part, and maybe others can help for this?

Comment (jiangliu) [reply storage/src/cache/cachedfile.rs:383]: Could you please help to open an issue for stargz and revert related change? So we could move on with non-stargz part.

Comment (kevinXYin) [reply storage/src/cache/cachedfile.rs:383]: yes , this should be better

**Comment by yqleng1987** [comment]
@kevinXYin , the code has been updated, so a new test job has been submitted. Please wait in patience. The test job url: [https://tone.openanolis.cn/ws/nrh4nnio/test_result/14180](https://tone.openanolis.cn/ws/nrh4nnio/test_result/14180)

**Comment by yqleng1987** [comment]
@kevinXYin , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14180>:white_check_mark: SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14180>:white_check_mark: SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14180>:white_check_mark: SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14180>:white_check_mark: SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14180>:x: FAIL</a></td>
    </tr>
</table>

Sorry, your test job failed. Please get the details in the link.

**Comment by yqleng1987** [comment]
@hsiangkao , the test job has been submitted. Please wait in patience. The test job url: [https://tone.openanolis.cn/ws/nrh4nnio/test_result/14199](https://tone.openanolis.cn/ws/nrh4nnio/test_result/14199)

**Comment by yqleng1987** [comment]
@hsiangkao , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14199>:white_check_mark: SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14199>:white_check_mark: SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14199>:white_check_mark: SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14199>:white_check_mark: SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14199>:white_check_mark: SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14199>:white_check_mark: SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/14199>:white_check_mark: SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

**Review by jiangliu — APPROVED** [review]

**Pull request merged**